pkg/cmd/repo/rename/rename.go

```diff
@@ -0,0 +1,113 @@
+package rename
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/internal/ghinstance"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type RenameOptions struct {
+	HttpClient  func() (*http.Client, error)
+	IO          *iostreams.IOStreams
+	Config      func() (config.Config, error)
+	oldRepoName string
+	newRepoName string
+}
+
+type renameRepo struct {
+	Owner      string
+	Repository string
+	Name       string `json:"name,omitempty"`
+}
+
+func NewCmdRename(f *cmdutil.Factory, runf func(*RenameOptions) error) *cobra.Command {
+	opts := &RenameOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+	}
+
+	cmd := &cobra.Command{
+		DisableFlagsInUseLine: true,
+
+		Use:   "rename <user/old_repo_name> <new_repo_name>",
```

Comment:
```suggestion
		Use:   "rename <repository> <name>",
```
I am wondering if we should allow the name to be blank and then interactively prompt the user to enter a name if one is not provided?

Reply:
I'm into that, but I think we might want to pause and think through some possible UX cases